app/src/main/java/org/fossasia/pslab/activity/SensorActivity.java

```diff
@@ -140,43 +131,49 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
         });
     }
 
-    private void populateSensors() {
-        ArrayList<Integer> data = new ArrayList<>();
-        dataName.clear();
-        dataAddress.clear();
-        try {
-            tvSensorScan.setText(getResources().getString(R.string.scanning));
-            data = i2c.scan(null);
-        } catch (IOException | NullPointerException e) {
-            e.printStackTrace();
+    private class PopulateSensors extends AsyncTask<Void, Void, Void> {
+        private ArrayList<Integer> data;
+
+        @Override
+        protected Void doInBackground(Void... voids) {
+            data = new ArrayList<>();
+            dataName.clear();
+            dataAddress.clear();
+            try {
+                data = i2c.scan(null);
+            } catch (IOException | NullPointerException e) {
+                e.printStackTrace();
+            }
+            return null;
         }
-        if (data != null) {
-            for (Integer myInt : data) {
-                if (myInt != null && sensorAddr.get(myInt) != null) {
-                    dataAddress.add(String.valueOf(myInt));
+
+        @Override
+        protected void onPostExecute(Void aVoid) {
+            super.onPostExecute(aVoid);
+
+            if (data != null) {
+                for (Integer myInt : data) {
+                    if (myInt != null && sensorAddr.get(myInt) != null) {
+                        dataAddress.add(String.valueOf(myInt));
+                    }
                 }
-            }
-            tvData = "";
+                tvData = "";
 
-            for (String s : dataAddress) {
-                tvData += s + ":" + sensorAddr.get(Integer.parseInt(s)) + "\n";
+                for (String s : dataAddress) {
+                    tvData += s + ":" + sensorAddr.get(Integer.parseInt(s)) + "\n";
+                }
+
+            } else {
+                tvData = getResources().getString(R.string.sensor_not_connected);
             }
 
-        }
-        else {
-            tvData = getResources().getString(R.string.sensor_not_connected);
-        }
+            for (int key : sensorAddr.keySet()) {
+                dataName.add(sensorAddr.get(key));
+            }
 
-        for (int key : sensorAddr.keySet()) {
-            dataName.add(sensorAddr.get(key));
+            tvSensorScan.setText(tvData);
+            adapter.notifyDataSetChanged();
+            buttonSensorAutoScan.setClickable(true);
         }
```

Comment:
Did you forget to remove this line as well? It might cause compilation error

Reply:
Why should '}' cause compilation error? It is the ending bracket of onPostExecute() method.

Reply:
My bad. It looks like the closing bracket for the for loop instead of the function.